tempodb/encoding/vparquet/wal_block.go

```diff
@@ -460,6 +463,8 @@ func (b *walBlock) FindTraceByID(ctx context.Context, id common.ID, _ common.Sea
 			}
 
 			r := parquet.NewReader(file)
+			defer r.Close()
```

Comment:
should we also defer close the file above? and all of the files returned from `page.file()` calls?

Reply:
`parquet.File` unfortunately does not implement a Close function, so we can not close it. I noticed that when I was figuring out what is going on and logged an issue - https://github.com/grafana/tempo/issues/2153

Reply:
Sounds interesting.

What do you think of wrapping the parquet.File with another type implements Close()? I feel like maybe that would be a more intuitive design. Also, it could later be used in other call sites theoretically (which would probably also need to close the file at some point).

Reply:
so like:
```
type pageFile struct {
  parquet.File
  os.File
}
```
then the caller can interact with the parquet.file and eventually close the os.file? i'm good on that

Reply:
Added closing the file opened in `file()`. It resulted in a little bit more changes that I anticipated, but we should close everything opened in the wal_block module now

Reply:
Ah, did all that work, but didnt call the finel `iterator.Close()`. Fixed that